### ISSUE TITLE
Optional keys (key?: type) and super types

### ISSUE THREAD

**Issue by AugustinLF** [facebook/flow#5795]
Hey,
I thought I finally got it with [subtypes, covariant and exact objects](https://github.com/facebook/flow/issues/4495), but I'm struggling with a case, where I'd want to initialise a form with my API response, which I believe is a super type of the form. 
```javascript
const response: {|+a: string|} = {a: 'aha'}
const form: {+a: string, +b?: ?string} = response
```
Flow complains on the b type. Since I can't mutate the objects, I struggle to understand what I'm doing wrong here.

Here's the [flow try](https://flow.org/try/#0MYewdgzgLgBATgUwgLhgbwD4GoCGrpwCWYA5hgL4wC86eMA5DgBY73kBQoksAZiHAFtUmXPihFSAGhhYARgH5U8gsTKUaiCEA). For information, I can't modify the response type, since it's relay generated type.

**Comment by AugustinLF** [comment]
Hmmm, I understand, but [this doesn't work either](https://flow.org/try/#0MYewdgzgLgBATgUwgLhgbwNQENXTgSzAHMBfGAXnRxgHIsALLGkgKFElgDMQ4BbVTNTyEiAGhgYARgH5U04cTKVEEIA) without an exact object type. Are you aware of any workaround, to make my use-case work? Moreover, I understand that the design of exact object types might not let me do that, but is there a "valid case" which makes my program buggy, or is this simply a limitation of flow?

**Comment by idiostruct** [comment]
Could `form` have extra properties or should it match the response?

**Comment by AugustinLF** [comment]
Form can have extra properties (it already does actually, it's just initialized with the response), it's only the response that I cannot modify, since it's generated by the `relay-compiler`.

**Comment by idiostruct** [comment]
Does spreading `res` into an object literal work for you needs and keep type safety?

**Comment by AugustinLF** [comment]
Yes [it does](https://flow.org/try/#0MYewdgzgLgBATgUwgLhgbwD4GoCGrpwCWYA5hgL4wC86eMA5DgBY73kBQoksAZiHAFtUaXPihFSAGhhYARgH5U8gsRKUaaAHTbEEckA)! It makes sense, thanks for the help, even I'd still like to know if my intuition fails me when it tells me that my initial example *should* work^^

**Comment by SamChou19815** [comment]
The example now works
